aptos-move/framework/aptos-framework/sources/genesis.move

```diff
@@ -315,21 +302,15 @@ module aptos_framework::genesis {
     /// Network address fields are a vector per account, where each entry is a vector of addresses
     /// encoded in a single BCS byte array.
     fun create_initialize_validators(aptos_framework: &signer, validators: vector<ValidatorConfiguration>) {
-        let i = 0;
-        let num_validators = vector::length(&validators);
-
         let validators_with_commission = vector::empty();
-
-        while (i < num_validators) {
+        vector::for_each_reverse(validators, |validator| {
```

Comment:
Why use for_each_reverse instead of for_each here?

Reply:
The loop is using `vector::pop_back(&mut validators)` so actually it iterates in the reverse order. In fact `for_each` is implemented with `for_each_reverse`([link](https://github.com/aptos-labs/aptos-core/blob/bd3eb542ef22a2067653417295a02a2fb571efd8/aptos-move/framework/move-stdlib/sources/vector.move#L243)). This is because of ownership. So `for_each_reverse` is more efficient.